src/main/java/org/itsallcode/openfasttrace/report/PlainTextReport.java

```diff
@@ -219,36 +223,119 @@ private void renderMaturity(final LinkedSpecificationItem item, final PrintStrea
         }
     }
 
+    private void renderFailureDetails(final PrintStream report)
+    {
+        this.trace.getDefectItems().stream() //
+                .sorted(LINKED_ITEM_BY_ID) //
+                .forEachOrdered(item -> renderItemDetails(report, item));
+    }
+
     private void renderAll(final PrintStream report)
     {
         this.trace.getItems().stream() //
                 .sorted(LINKED_ITEM_BY_ID) //
-                .forEachOrdered(item -> renderItemDetails(item, report));
+                .forEachOrdered(item -> renderItemDetails(report, item));
     }
 
-    private void renderItemDetails(final LinkedSpecificationItem item, final PrintStream report)
+    private void renderItemDetails(final PrintStream report, final LinkedSpecificationItem item)
     {
-        renderItemSummary(item, report);
+        renderItemSummary(report, item);
+        renderDescription(report, item);
+        renderLinks(report, item);
+        renderTags(report, item);
+        renderItemDetailsEnd(report);
+    }
 
+    private void renderEmptyItemDetailsLine(final PrintStream report)
+    {
+        report.print("|");
+        report.print(this.newline);
+    }
+
+    private void renderDescription(final PrintStream report, final LinkedSpecificationItem item)
+    {
         final String description = item.getDescription();
         if (description != null && !description.isEmpty())
         {
-            report.print("|");
-            report.print(this.newline);
+            renderEmptyItemDetailsLine(report);
             for (final String line : description.split(Newline.anyNewlineReqEx()))
             {
                 report.print("| ");
                 report.print(line);
                 report.print(this.newline);
             }
-            report.print("|");
-            report.print(this.newline);
+            ++this.nonEmptySections;
         }
+    }
+
+    // [impl->dsn~reporting.plain-text.link-details~1]
+    private void renderLinks(final PrintStream report, final LinkedSpecificationItem item)
+    {
+
+        final Map<LinkStatus, List<LinkedSpecificationItem>> links = item.getLinks();
+        if (!links.isEmpty())
+        {
+            renderEmptyItemDetailsLine(report);
+            renderOrderedLinks(report, links);
+            ++this.nonEmptySections;
+        }
+    }
+
+    private void renderOrderedLinks(final PrintStream report,
+            final Map<LinkStatus, List<LinkedSpecificationItem>> links)
+    {
+        final List<TracedLink> tracedLinks = flattenTracedLinks(links);
+        tracedLinks.stream() //
+                .sorted((a, b) -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)